Automattic/wp-calypso
# Stats: Switch between weeks, days, months and years in Post stats page

closes #10344 

In this PR, I update the Post Stats Page and add tabs to switch the post views chart between days, weeks, months and years.

<img width="733" alt="screen shot 2017-01-04 at 14 03 57" src="https://cloud.githubusercontent.com/assets/272444/21643086/acd2f1c0-d286-11e6-8864-fe795787facd.png">

To achieve this results, I followed these steps:

 - Extracted the UI part of the former `StatsSummaryChart` to its own sub-component
 - This sub-component is used in two places: 
     - StatsPostSummary to show the chart for the selected period (days, weeks, ....)
     - StatsVideoSummary to mimic the old StatsSummaryChart behaviour for videos.
 - Using QueryPostStats to retrieve the stats for the StatsPostSummary component.

**Nice to have**

 - We could use the same switching mechanism for Video Stats as well!

**Testing instructions**

 - Navigate to the Post Stats Page: `/stats/post/$postId/$site`
 - Switch between periods (days, weeks ....)
 - Try clicking on the bars, the displayed views count should be updated in consequence
